packages/ui/src/memberships/components/MemberProfile/MemberRoleToggle.tsx

```diff
@@ -1,69 +1,76 @@
-  SidePaneRow,
-  SidePaneTable,
-  SidePaneText,
-} from '../../../common/components/SidePane'
-import { BorderRad, Colors } from '../../../common/constants'
-import { Member } from '../../types'
+import { UnknownAccountInfo } from '@/accounts/components/UnknownAccountInfo'
+import { BlockTime } from '@/common/components/BlockTime'
+import { ButtonGhost, ButtonsGroup } from '@/common/components/buttons'
+import { ToggleableItem, ToggleButton } from '@/common/components/buttons/Toggle'
+import { Arrow } from '@/common/components/icons'
+import { SidePaneColumn, SidePaneLabel, SidePaneRow, SidePaneTable, SidePaneText } from '@/common/components/SidePane'
+import { BorderRad, Colors } from '@/common/constants'
+import { formatTokenValue } from '@/common/model/formatters'
+import { Member } from '@/memberships/types'
+import { workerRoleTitle } from '@/working-groups/helpers'
+import { WorkerWithDetails } from '@/working-groups/types'
 
-export interface MemberRoleToggleProps {
+export interface Props {
   member: Member
-  role: string
+  role: WorkerWithDetails
 }
 
-export const MemberRoleToggle = ({ member, role }: MemberRoleToggleProps) => {
+export const MemberRoleToggle = ({ role }: Props) => {
   return (
     <RoleToggle absoluteToggle>
       {(isOpen) => {
         return (
           <MemberRoleToggleContainer>
             <MemberRoleTitleContainer>
-              <h5>{role}</h5>
+              <h5>{workerRoleTitle(role)}</h5>
             </MemberRoleTitleContainer>
             {isOpen && (
               <MemberRoleTableContainer>
                 <MemberRoleTable>
                   <SidePaneRow>
                     <SidePaneLabel text="Hired" />
-                    <SidePaneColumn>Add block component here</SidePaneColumn>
+                    <SidePaneColumn>
+                      <BlockTime block={role.hiredAtBlock} />
+                    </SidePaneColumn>
                   </SidePaneRow>
                   <SidePaneRow>
                     <SidePaneLabel text="Reward" />
                     <SidePaneColumn>
-                      <SidePaneText>13 923 JOY / 3600 blocks</SidePaneText>
+                      <SidePaneText>{formatTokenValue(role.rewardPerBlock)} JOY / 3600 blocks</SidePaneText>
                     </SidePaneColumn>
                   </SidePaneRow>
                   <SidePaneRow>
                     <SidePaneLabel text="Earned total" />
                     <SidePaneColumn>
-                      <SidePaneText>20 923M JOY </SidePaneText>
+                      <SidePaneText>{formatTokenValue(role.earnedTotal)} JOY</SidePaneText>
                     </SidePaneColumn>
                   </SidePaneRow>
                   <SidePaneRow>
                     <SidePaneLabel text="Earned in 24h" />
                     <SidePaneColumn>
-                      <SidePaneText> {member.id} JOY (id for test) </SidePaneText>
+                      <SidePaneText>{formatTokenValue(1000)} JOY</SidePaneText>
```

Comment:
We have a dedicated component for this: `<TokenValue />`

```suggestion
                      <SidePaneText><TokenValue value="1000" /></SidePaneText>
```

And for other token values in this file.